### PR TITLE
Cleaned up multiple inheritance in base bead

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
1. Changed from multiple inheritance to composition using name and identity objects
2. Replaced bead and basebead variable names to more explicit names
3. Replaced prepended underscore to post pended underscore to satisfy the standard.

### CODE REVIEW

File: include/votca/csg/basebead.h [hunk 1]
```diff
@@ -110,40 +128,46 @@ class BaseBead : public TopologyItem,
...
+  bead_position_set_ = true;
+  bead_position_ = bead_position;
 }
 
 inline const vec &BaseBead::getPos() const {
-  assert(_bPos);
-  return _pos;
-}
-}
+  assert(bead_position_set_);
```
Comment (JensWehner) [inline include/votca/csg/basebead.h]: make it `assert(bead_position_set_ & "Position not set")` an assert without a message is really hard to debug

File: include/votca/csg/basebead.h [hunk 2]
```diff
@@ -110,40 +128,46 @@ class BaseBead : public TopologyItem,
    * \return reference to position
    */
   virtual vec &Pos() {
-    assert(_bPos);
-    return _pos;
+    assert(bead_position_set_);
```
Comment (JensWehner) [inline include/votca/csg/basebead.h]: see below

File: include/votca/csg/bead.h [hunk 3]
```diff
@@ -186,51 +186,51 @@ class Bead : public BaseBead {
    * \return reference to velocity
    */
   vec &Vel() {
-    assert(_bVel);
-    return _vel;
+    assert(bead_velocity_set_);
```
Comment (JensWehner) [inline include/votca/csg/bead.h]: see above

File: include/votca/csg/bead.h [hunk 4]
```diff
@@ -186,51 +186,51 @@ class Bead : public BaseBead {
...
   }
 
   /**
    * direct access (read/write) to orientation u of the bead
    * \return reference to u
    */
   vec &U() {
-    assert(_bU);
-    return _u;
+    assert(bU_);
```
Comment (JensWehner) [inline include/votca/csg/bead.h]: again

File: include/votca/csg/bead.h [hunk 5]
```diff
@@ -276,7 +276,7 @@ class Bead : public BaseBead {
    * If it is a mapped beads, returns te bead id the cg bead was created from
    * \return vector of bead ids of reference atoms
    */
-  std::vector<int> &ParentBeads() { return _parent_beads; };
+  std::vector<int> &ParentBeads() { return parent_beads_; };
```
Comment (JensWehner) [inline include/votca/csg/bead.h]: why not const and if not create also a const version

File: include/votca/csg/bead.h [hunk 6]
```diff
@@ -308,49 +312,49 @@ class Bead : public BaseBead {
...
-    _bF = false;
+    bead_position_set_ = false;
+    bead_velocity_set_ = false;
+    bU_ = false;
+    bV_ = false;
+    bW_ = false;
+    bead_force_set_ = false;
   }
 
   void *_userdata;
```
Comment (JensWehner) [inline include/votca/csg/bead.h]: where is this used? The type information has been cased away

Comment (JensWehner) [reply include/votca/csg/bead.h]: @junghans do you have an idea about this. I am not good with pointers

**Review by JensWehner — CHANGES_REQUESTED** [review]
good step in the right direction

File: include/votca/csg/bead.h (hunk 6 above)
Comment (JoshuaSBrown) [reply include/votca/csg/bead.h]: I keep seeing this user data pointer in other classes but it still isn't clear to me what exactly they are used for.

Comment (JoshuaSBrown) [reply include/votca/csg/bead.h]: I can't find it anywhere so I guess I'll pull it out.

Comment (JoshuaSBrown) [reply include/votca/csg/bead.h]: The same is true for Options it does not appear to be used anywhere...

**Comment by JoshuaSBrown** [comment]
I removed the options and user data from the beads, nothing is ever added to these attributes they are just passed around. I imagine the intention was that if someone ever wanted to extend the functionality without making changes to the underlying structures they could used these attributes.

**Comment by JoshuaSBrown** [comment]
@JensWehner do you want to check this once more before I merge?

**Pull request merged**